Chore / maintenance 2023 q4

royschut opened the pull request:
## Maintenance 2023-Q4
- Perform `yarn upgrade`
- Replace `rollup-plugin-postcss` with `rollup-plugin-import-css`. The former used an older version of `loader-utils` that has **critical** issues. 
- Add resolutions for `minimist` and `minimatch` for **critical** issues that couldn't be fixed by updates
- Update all eslint deps to their latest version
- Clean up unnecessary deps: `eslint-watch`, `eslint-plugin-codeceptjs` and `rimraf`
- Add the example project to the readme
- Update `vite` in the example project. All it's deps are fully up-to-date now.

Only 2 moderate issues left, that could only be fixed by resolutions, which is perhaps overkill.

### Notes
- I'm tempted to clean up the eslint config and plugins in general, since we only have one `tsx` and one `css` file in the project. Perhaps next sprint. Any thoughts? Why should we need so much?

MelissaDTH reviewed (commented):
It looks good! I have two questions:
1. What is the reason for the file naming: lib/assets/index-UHUyKXD4.css?
2. Have you been able to test this?

ChristiaanScheermeijer reviewed (commented):
All looks good, but we need to change the CSS import behavior (and add this to the readme).

Since we're not transforming the CSS file, we probably better just copy the CSS file to the lib folder on build. Then the consumer can import the CSS and make overrides or customizations more easily.

We should also try removing the lib folder from git. Not sure why this ended up there. We already do a `prepublishOnly` script that builds the library before packaging.

royschut commented:
> It looks good! I have two questions:
> 
> 1. What is the reason for the file naming: lib/assets/index-UHUyKXD4.css?
> 2. Have you been able to test this?

1. That was because of the new rollup plugin that handles css bundling differently. But this is obsolete now, now that we just copy the css file, as Christiaan suggested.
2. Yes, but from the example project only. We'd have to see if it also works when published to npm.


> All looks good, but we need to change the CSS import behavior (and add this to the readme).
> 
> Since we're not transforming the CSS file, we probably better just copy the CSS file to the lib folder on build. Then the consumer can import the CSS and make overrides or customizations more easily.
> 
> We should also try removing the lib folder from git. Not sure why this ended up there. We already do a `prepublishOnly` script that builds the library before packaging.

Thanks! I removed the lib folder from git. I also added a rollup plugin to copy the css file.
While trying to make it work in the example project, first I tried to add TileSlider as a dependency from the lib folder, but I couldn't succeed (import errors), so I kept this as is, and just imported the css file from lib only. I've also updated the readme. Do you think this is sufficient? @ChristiaanScheermeijer

ChristiaanScheermeijer reviewed (approved):
LGTM! You can test a install by running these steps:

```shell
$ yarn pack
> success Wrote tarball to "<path-to-tarball>".

$ cd /path/to/project
$ yarn add <path-to-tarball>
$ yarn start
```

The above will remove all dependencies and result in the same package when installed via the NPM registry.

royschut commented:
> LGTM! You can test a install by running these steps:

Nice! I was looking for that. Thanks
